src/pykeen/sampling/negative_sampler.py

```diff
@@ -52,15 +55,63 @@ def __init__(
         self.filterer = filterer_resolver.make(
             filterer,
             pos_kwargs=filterer_kwargs,
-            triples_factory=triples_factory,
+            mapped_triples=triples_factory.mapped_triples,
         ) if filtered else None
 
-    @classmethod
-    def get_normalized_name(cls) -> str:
```

Comment:
was there a reason this got deleted?

Reply:
This is the same functionality as provided by the `class-resolver` library.

Reply:
True, but this function means that wherever there's an instance or class we directly have access. I think it's worth having the convenience

Reply:
Okay - I just found it in this class though. 

If we bring it back, we should replace it by `negative_sampler_resolver.normalize_cls(negative_sampler_cls)` to ensure it is consistent with it (and to avoid duplicating functionality from `class-resolver`).

Reply:
That will cause circular dependency, though

Reply:
Does it? The negative sampler resolver should reside in the negative sampler module anyway?

Reply:
The negative sampler resolver lives in the init file for pykeen.sampling, and the base class is in its own file. For the `__subclasses__` list to work, the other ones already have to be imported, so they can't live in the same places

Reply:
What about (re-)importing `negative_sampler_resolver` within `get_normalized_name`? This shouldn't be a performance issue since this method shouldn't be called frequently.

Reply:
I don't think this is actually relevant for this PR? Maybe we can open an issue and make a more general solution that isn't just for negative samplers, but all of the other classes that have this `get_normalized_name` class method

Reply:
Alright, feel free to revert.

I thought I have not seen this `get_normalized_name` anywhere else though.